docs/content/reference/server-configuration.md

```diff
@@ -1,52 +1,208 @@
+pre: "<i class='fa fa-cogs'></i> "
 ---
+{{< table_of_contents >}}
 
-## Configuring the Monkey Island
+## Configuration
 
-The Monkey Island Server is configured through the `server_config.json` file.
+<!--
+This documentation was autogenerated by passing the plugin's config-schema.json
+through https://github.com/adobe/jsonschema2md. It was then modified by hand to
+remove extraneous information.
+-->
 
-{{% notice info %}}
-Refer to the [setup guides](../../setup/) to learn how to use
-the `server_config.json` file for each deployment.
-{{% /notice %}}
+| Property                             | Type      | Required | Nullable       |
+| :----------------------------------- | :-------- | :------- | :------------- |
+| [data\_dir](#data_dir)               | `string`  | Optional | cannot be null |
+| [log\_level](#log_level)             | `string`  | Optional | cannot be null |
+| [mongodb](#mongodb)                  | `object`  | Optional | cannot be null |
+| [ssl\_certificate](#ssl_certificate) | `object`  | Optional | cannot be null |
+| [island\_port](#island_port)         | `integer` | Optional | cannot be null |
 
-### Creating a configuration file
+### data\_dir
+
+The directory where the Island will store runtime artifacts.
+
+`data_dir`
+
+* is optional
+
+* Type: `string`
+
+* cannot be null
+
+#### data\_dir Constraints
+
+The value of this string must be a valid path.
+
+#### data\_dir Default Value
+
+The default value for Windows is:
+
+```json
+"%AppData%\\monkey_island"
+```
+
+The default value for Linux is:
+
+```json
+"$HOME/.monkey_island"
+```
+
+### log\_level
+
+The threshold for the Island logger.
+
+`log_level`
+
+* is optional
+
+* Type: `string`
+
+* cannot be null
+
+#### log\_level Default Value
+
+The default value is:
+
+```json
+"INFO"
+```
+
+### mongodb
+
+The MongoDB configuration for the Island server.
+
+`mongodb`
+
+* is optional
+
+* Type: `object`
+
+* cannot be null
+
+#### mongodb Properties
+**_start\_mongodb_**
+
+If enabled, the MongoDB server will be started automatically with the Island.
+
+`start_mongodb`
+
+* is optional
+
+* Type: `boolean`
+
+* cannot be null
+
+_start\_mongodb Default Value_
+
+The default value is:
+
+```json
+true
+```
+
+#### mongodb Default Value
+
+The default value is:
 
-Here's an example `server_config.json` with all options specified:
 ```json
 {
-  "island_port": 443,
-  "log_level": "DEBUG",
-  "ssl_certificate": {
-    "ssl_certificate_file": "<PATH_TO_CRT_FILE>",
-    "ssl_certificate_key_file": "<PATH_TO_KEY_FILE>"
-  },
-  "mongodb": {
-    "start_mongodb": true
-  },
-  "data_dir": "/monkey_island_data"
+  "start_mongodb": true
 }
 ```
 
-Only relevant options can be specified, for example:
+### ssl\_certificate
+
+The SSL certificates configuration for the Island server.
+
+`ssl_certificate`
+
+* is optional
+
+* Type: `object`
+
+* cannot be null
+
+#### ssl\_certificate Properties
+**_ssl\_certificate\_file_**
```

Comment:
I think we should nix the header at this level and do `#### ssl_certificate_file` and `#### ssl_certificate_key_file`. At the top level, there's no header for "properties". In other words, Under `## Configuration` there is not `### Configuration Properties`.

Reply:
Same for mongo.

Reply:
Looks mostly fine to me except that "start_mongodb" and "mongodb Default Value" are both h4. 
![image](https://github.com/user-attachments/assets/7cb9e575-84e6-43d0-ad06-491279472046)

Reply:
You could omit "mongodb Default value". All of `mongodb`'s properties have default values, so you can extrapolate the default value of the whole object from there. You could also provide an example of the whole file in a different subsection at the same level as `### Configuration Properties`.